src/gui/dialogs/select_orb_colors.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ Part of the Battle for Wesnoth Project http://www.wesnoth.org/
+ 
+ This program is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2 of the License, or
+ (at your option) any later version.
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY.
+ 
+ See the COPYING file for more details.
+ */
+
+#define GETTEXT_DOMAIN "wesnoth-lib"
+
+#include "select_orb_colors.hpp"
+
+#include "gui/auxiliary/event/dispatcher.hpp"
+#include "gui/auxiliary/find_widget.tpp"
+#include "gui/auxiliary/iterator/walker.hpp"
+#include "gui/widgets/button.hpp"
+#include "gui/widgets/grid.hpp"
+#include "gui/widgets/settings.hpp"
+#include "gui/widgets/toggle_button.hpp"
+#include "gui/widgets/window.hpp"
+
+#include "preferences.hpp"
+#include "game_config.hpp"
+
+#include <boost/bind.hpp>
+
+namespace gui2 {
+REGISTER_DIALOG(select_orb_colors);
+
+tselect_orb_colors::tselect_orb_colors()
+	: show_unmoved_(preferences::show_unmoved_orb())
+	, show_partial_(preferences::show_partial_orb())
+	, show_moved_(preferences::show_moved_orb())
+	, show_ally_(preferences::show_allied_orb())
+	, show_enemy_(preferences::show_enemy_orb())
+	, unmoved_(preferences::unmoved_color())
+	, partial_(preferences::partial_color())
+	, moved_(preferences::moved_color())
+	, ally_(preferences::allied_color())
+	, enemy_(preferences::enemy_color())
+{
+}
+
+void tselect_orb_colors::pre_show(CVideo&, twindow& window)
+{
+	setup_orb_group("unmoved", show_unmoved_, unmoved_, window);
+	setup_orb_group("partial", show_partial_, partial_, window);
+	setup_orb_group("moved", show_moved_, moved_, window);
+	setup_orb_group("ally", show_ally_, ally_, window);
+	setup_orb_group("enemy", show_enemy_, enemy_, window);
+	
+	tbutton& reset = find_widget<tbutton>(&window, "orb_defaults", false);
+	event::connect_signal_mouse_left_click(reset, boost::bind(
+		&tselect_orb_colors::handle_reset_click,
+		this, boost::ref(window)
+	));
+}
+
+void tselect_orb_colors::display(CVideo& video)
+{
+	tselect_orb_colors dialog;
+	if(dialog.show(video)) {
```

Comment:
Such a thing should be handled in the post_show function. The display() function isn't supposed to be implemented outside the header.